### PR TITLE
Experience Platform Connector extension

### PR DESCRIPTION
**Pull request by jcalcaben**
## Description

Creates the Experience Platform Connection extension package.

Also adds a SEARCHBAR_REQUEST event dispatch to the search barch autocomplete talon. 

## Related Issue

Closes PWA-2743

## Acceptance

Any core developers

### Verification Stakeholders

Any core developers

## Verification Steps

See Jira Issue

## Checklist

-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.

### CODE REVIEW

File: packages/extensions/experience-platform-connector/src/handlers/__tests__/signIn.spec.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+            publish: {
+                signIn: jest.fn()
+            }
+        };
+
+        const mockEvent = {
+            type: 'USER_SIGN_IN',
+            payload: {
+                email: 'doctor.strange@fake.email',
+                // RWTODO: confirm firstname/lastname property casing
```
Comment (anthoula) [inline packages/extensions/experience-platform-connector/src/handlers/__tests__/signIn.spec.js:37]: Is this TODO still valid?

File: packages/extensions/experience-platform-connector/src/handlers/addToCart.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+                  valueLabel: selectedOptions.value
+              }
+          ]
+        : null;
+
+    const cartItemContext = {
+        id: cartId,
+        prices: {
+            subtotalExcludingTax: {
+                value: priceTotal * quantity,
```
Comment (anthoula) [inline packages/extensions/experience-platform-connector/src/handlers/addToCart.js:29]: Interesting that prices are calculated here. Is this value not available elsewhere?

File: packages/extensions/experience-platform-connector/src/handlers/shoppingCartPageView.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+const canHandle = event => event.type === 'CART_PAGE_VIEW';
+
+const handle = (sdk, event) => {
+    const { payload } = event;
+
+    const { cart_id: id, products } = payload;
+
+    const cartContext = {
+        id: id,
```
Comment (anthoula) [inline packages/extensions/experience-platform-connector/src/handlers/shoppingCartPageView.js:11]: Optional suggestion for this and other instances in PR using the same key: value pair
```suggestion
        id,
```

**Review by anthoula — APPROVED** [review]
@jcalcaben - Great work on this extension! A few minor feedbacks that are totally optional, and clarifying question

File: packages/extensions/experience-platform-connector/src/handlers/addToCart.js (hunk 2 above)
Comment (jcalcaben) [reply packages/extensions/experience-platform-connector/src/handlers/addToCart.js:29]: none that I could find

**Comment by dpatil-magento** [comment]
run cypress

**Comment by dpatil-magento** [comment]
run cypress @search

**Comment by dpatil-magento** [comment]
run lighthouse-mobile

**Review by dpatil-magento — APPROVED** [review]

**Comment by dpatil-magento** [comment]
QA Approved

**Pull request merged**